hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -70,25 +72,32 @@
     AbstractHoodieWriteClient<T, JavaRDD<HoodieRecord<T>>, JavaRDD<HoodieKey>, JavaRDD<WriteStatus>> {
 
   private static final Logger LOG = LogManager.getLogger(SparkRDDWriteClient.class);
+  protected final transient HoodieObservabilityMetrics observabilityMetrics;
 
   public SparkRDDWriteClient(HoodieEngineContext context, HoodieWriteConfig clientConfig) {
-    super(context, clientConfig);
+    this(context, clientConfig, Option.empty());
   }
 
   @Deprecated
   public SparkRDDWriteClient(HoodieEngineContext context, HoodieWriteConfig writeConfig, boolean rollbackPending) {
-    super(context, writeConfig);
+    this(context, writeConfig, rollbackPending, Option.empty());
   }
 
   @Deprecated
   public SparkRDDWriteClient(HoodieEngineContext context, HoodieWriteConfig writeConfig, boolean rollbackPending,
                              Option<EmbeddedTimelineService> timelineService) {
     super(context, writeConfig, timelineService);
+    this.observabilityMetrics = (HoodieObservabilityMetrics) Registry.getRegistry(
+        HoodieObservabilityStat.OBSERVABILITY_REGISTRY_NAME, HoodieObservabilityMetrics.class.getName());
+    observabilityMetrics.registerWithSpark(context, config);
   }
 
   public SparkRDDWriteClient(HoodieEngineContext context, HoodieWriteConfig writeConfig,
                              Option<EmbeddedTimelineService> timelineService) {
     super(context, writeConfig, timelineService);
+    this.observabilityMetrics = (HoodieObservabilityMetrics) Registry.getRegistry(
```

Comment:
There is an initialize function where other metric registries are initialized. Better to move this code there.

Reply:
Moved.